pkg/controller/identity/iampolicy/controller.go

```diff
@@ -275,3 +282,44 @@ func (e *external) deleteNonDefaultVersions(ctx context.Context, policyArn strin
 
 	return nil
 }
+
+// ListPoliciesPages paginates ListPolicies
+func (e *external) ListPoliciesPages(ctx context.Context, input *awsiam.ListPoliciesInput, fn func(*awsiam.ListPoliciesOutput, bool) bool) error {
+	for {
+		output, err := e.client.ListPolicies(ctx, input)
+		if err != nil {
+			return err
+		}
+		lastPage := awsclient.StringValue(output.Marker) == ""
+		if !fn(output, lastPage) || lastPage {
+			break
+		}
+		input.Marker = output.Marker
+	}
+	return nil
+}
+
+// getPolicyArnByName will paginate through all policies to retrieve by name instead of arn
+func (e *external) getPolicyArnByName(ctx context.Context, policyName string) (*string, error) {
+
+	input := &awsiam.ListPoliciesInput{Scope: awsiamtypes.PolicyScopeTypeLocal}
```

Comment:
add path here, ref. performance? I think the path is immutable, so I'm not sure what's best to do if we have another policy with he same name and a different path, but we should at least think about the behavior if you create

* two policies in with different contents and same name/path
* two policies with different paths and same name

I think at least the second case might be confusing with this change?